apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryCommand.Data.cs

```diff
@@ -15,62 +15,240 @@
+        public override void Cancel()
+        {
+            SynchronousCancellationTokenSource.Cancel();
+        }
+
         /// <inheritdoc />
-        protected override DbParameterCollection DbParameterCollection { get; }
+        public override object ExecuteScalar() => ExecuteScalarAsync(SynchronousCancellationTokenSource.Token)
+            .ResultWithUnwrappedExceptions();
+
+        /// <summary>
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public override Task<object> ExecuteScalarAsync(CancellationToken cancellationToken)
+            => ExecuteScalarAsync<object>(cancellationToken);
+
+        /// <summary>
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<T> ExecuteScalarAsync<T>(CancellationToken cancellationToken)
+        {
+            using (var reader = await ExecuteReaderAsync(cancellationToken)
+                .ConfigureAwait(false))
+            {
+                if (await reader.ReadAsync(cancellationToken).ConfigureAwait(false) && reader.HasRows &&
+                    reader.FieldCount > 0)
+                    return reader.GetFieldValue<T>(0);
+            }
+            return default(T);
+        }
+
         /// <inheritdoc />
-        protected override DbTransaction DbTransaction { get; set; }
+        public override int ExecuteNonQuery()
+            => ExecuteNonQueryAsync(SynchronousCancellationTokenSource.Token).ResultWithUnwrappedExceptions();
+
         /// <inheritdoc />
-        public override bool DesignTimeVisible { get; set; }
+        public override Task<int> ExecuteNonQueryAsync(CancellationToken cancellationToken)
+        {
+            AssertCommandOptions();
+            return CommandOptions.ExecuteNonQueryAsync(BigQueryConnection, cancellationToken);
+        }
 
         /// <inheritdoc />
-        protected override DbParameter CreateDbParameter()
+        public override void Prepare()
         {
-            throw new NotImplementedException();
+            //BigQuery does not support preoptimized queries nor 2 phase commit transactions.
+            //Note: We cannot throw here because that will break some datatable scenarios.
         }
 
         /// <inheritdoc />
+        protected override DbParameter CreateDbParameter() => new BigQueryParameter();
+
+        /// <summary>
+        /// Sends the command to Cloud BigQuery and builds a <see cref="BigQueryDataReader"/>.
+        /// </summary>
+        public new Task<BigQueryDataReader> ExecuteReaderAsync()
+            => ExecuteReaderAsync(CommandBehavior.Default, CancellationToken.None);
+
+        /// <summary>
+        /// Sends the command to Cloud BigQuery and builds a <see cref="BigQueryDataReader"/>.
+        /// </summary>
+        public new async Task<BigQueryDataReader> ExecuteReaderAsync(CommandBehavior behavior,
```

Comment:
yes poor line break choice.  when I fix these manually, they wont get auto-changed.  This happens if I let something past the 120 guideline and the autoformat decides whats best.

Reply:
i thought we agreed on 120 chars?  I've tried to make it more readable with a better break, but there is a balance here so that it can fit properly in some scenarios.

Reply:
I use 120 as a guideline, but where things look significantly better by breaking that a bit, I'll generally do so.